src/paginate.ts

```diff
@@ -39,13 +39,14 @@ export { FilterOperator, FilterSuffix }
 export class Paginated<T> {
     data: T[]
     meta: {
-        itemsPerPage: number
-        totalItems: number
-        currentPage: number
-        totalPages: number
+        page: {
+            number: number
+            size: number
+            totalItems: number
+            totalPages: number
+        }
         sortBy: SortBy<T>
-        searchBy: Column<T>[]
-        search: string
+        search: { query: string; fields: Column<T>[] }
```

Comment:
`@search`

Reply:
In meta I opted to leave it as `search` as it may contain any key we like. Having the @ only where it was functionally required in the built links felt sufficient for me. Let me know, I have no strong feelings :)

Reply:
I'm leaving it on just `search` because otherwise we need `["@search"]` which felt inconvenient